src/main/java/org/veupathdb/service/eda/us/model/TabularDataWriter.java

```diff
@@ -0,0 +1,42 @@
+package org.veupathdb.service.eda.us.model;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+
+public interface TabularDataWriter {
+
+  void write(String... column) throws IOException;
+
+  void nextRecord() throws IOException;
+
+  class TsvFormatter implements TabularDataWriter {
+    private static final byte[] TAB_BYTES = "\t".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] NEW_LINE_BYTES = System.lineSeparator().getBytes(StandardCharsets.UTF_8);
+
+    private boolean firstColumnInRow = true;
+    private OutputStream outputStream;
+
+    public TsvFormatter(OutputStream outputStream) {
+      this.outputStream = outputStream;
+    }
+
+    @Override
+    public void write(String... columns) throws IOException {
```

Comment:
add a comment explaining why u are writing bytes not strings?

Reply:
Will do, I actually tried wrapping the ZipOutputStream in an OutputStreamWriter which allows us to write strings instead of bytes and ran into issues. I'll leave a comment.